src/app/users/users.service.ts

```diff
@@ -48,6 +49,36 @@ export class UsersService {
     return this.http.get(`/users/${userId}`);
   }
 
+  /**
+   * Confirm Change Password of Users
+   * @param controlNameToCompare
+   */
+  confirmPassword(controlNameToCompare: string): ValidatorFn {
+    return (c: AbstractControl): ValidationErrors|null => {
+        if (c.value == null || c.value.length === 0) {
+            return null;
+        }
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
+
+  /**
+   * Change User Password
+   * @param userId User Id of users
+   * @param password New Password of the user
+   * @returns {Observable<any>}
+   */
+  changePassword(userId: string, passwordObj: any) {
+    return this.http.put(`/users/${userId}`, passwordObj);
+  }
+
```

Comment:
why are the same functions in 2 services?

Reply:
What is the other service this function being repeated?